src/client/Inode.cc

```diff
@@ -783,6 +783,7 @@ void Inode::mark_caps_dirty(int caps)
     iget();
   dirty_caps |= caps;
   client->get_dirty_list().push_back(&dirty_cap_item);
+  client->cap_delay_requeue(this);
```

Comment:
How did this come up? is it a separate bug? If we don't have a specific issue pointing to this, I don't think it's a heuristic we should change.

Reply:
This is just one perf improvement fixing. In some cases the MDS will try to recall the caps from clients, while in client side it's possible that the dirty caps will be held for a very long time without getting a chance to be flushed if no `check_caps()` is called for the Inodes.

Today I checked the kclient, it's already done this. I am okay to remove this in this PR if it makes no sense.

Reply:
Okay, if the kclient is already behaving this way it's probably a good change to make. I was just a bit confused because I thought it was going to be part of the mdlog sync flushing fix and didn't have context for a specific problem it could cause. :)